src/Core/Foundation/Version/Version.php

```diff
@@ -0,0 +1,255 @@
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShop\PrestaShop\Core\Foundation\Version;
+
+use PrestaShop\PrestaShop\Core\Foundation\Version\Exception\InvalidVersionException;
+
+/**
+ * Class responsible of managing the right version of Shop
+ * for every internal/external services.
+ */
+class Version
+{
+    const STRING = 0;
+    const INTEGER = 1;
+
+    /**
+     * Full version name.
+     *
+     * @var string
+     */
+    private $fullVersion;
+
+    /**
+     * Major version name.
+     *
+     * @var string
+     */
+    private $majorVersionString;
+
+    /**
+     * Major version.
+     *
+     * @var int
+     */
+    private $majorVersion;
+
+    /**
+     * Minor version.
+     *
+     * @var int
+     */
+    private $minorVersion;
+
+    /**
+     * Release version.
+     *
+     * @var int
+     */
+    private $releaseVersion;
+
+
+    public function __construct($version, $majorVersionString, $majorVersion, $minorVersion, $releaseVersion)
+    {
+        $this->version = $version;
+        $this->majorVersionString = $majorVersionString;
+        $this->majorVersion = $majorVersion;
+        $this->minorVersion = $minorVersion;
+        $this->releaseVersion = $releaseVersion;
+    }
+
+    /**
+     * Returns the current PrestaShop version which is hardcoded in \AppKernel::VERSION.
+     *
+     * @return string For example "1.7.4.0"
+     */
+    public function getFullVersion()
+    {
+        return $this->fullVersion;
+    }
+
+    /**
+     * Returns the current PrestaShop major version as a string.
+     *
+     * @return string For example "1.7"
+     */
+    public function getStringMajorVersion()
+    {
+        return $this->majorVersionString;
+    }
+
+    /**
+     * Returns the current PrestaShop major version as an integer.
+     *
+     * @return int For example 17
+     */
+    public function getMajorVersion()
+    {
+        return $this->majorVersion;
+    }
+
+    /**
+     * Returns the current PrestaShop minor version.
+     *
+     * @return int
+     */
+    public function getMinorVersion()
+    {
+        return $this->minorVersion;
+    }
+
+    /**
+     * Returns the current PrestaShop release version
+     *
+     * @return int
+     */
+    public function getReleaseVersion()
+    {
+        return $this->releaseVersion;
+    }
+
+    /**
+     * Returns if the current PrestaShop version is greater than the provided version.
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
```

Comment:
do you mean `InvalidVersionException` instead of `InvalidArgumentException`?

Reply:
@sarjon Ups, I will fix it in the evening, thank you!